client/src/www/model/platform_error.ts

```diff
@@ -216,14 +193,31 @@ export class PlatformError extends CustomError {
     }
     return result;
   }
+}
 
-  /**
-   * Returns a JSON string of this error with all details and causes.
-   * @returns {string} The JSON string representing this error.
-   */
-  toJSON(): string {
-    const errRawObj = convertPlatformErrorToRawErrorObject(this);
-    return JSON.stringify(errRawObj);
+/**
+ * ipcToAppError converts an Error returned from the MethodChannel IPC to an application error.
+ * MethodChannel errors encode its information as a JSON object in the Error message.
+ */
+export function ipcToAppError(ipcError: Error): CustomError {
```

Comment:
We may also mark the `parseFrom` `@deprecated` as this is the new method to use.